src/sqlfluff/utils/analysis/select_crawler.py

```diff
@@ -177,40 +178,32 @@ def crawl_sources(
             "select_statement",
             "values_clause",
         ]
-        for event, path in SelectCrawler.visit_segments(
-            segment, recurse_into=recurse_into
+        for seg in segment.recursive_crawl(
+            *types, recurse_into=False, allow_self=False
         ):
-            seg = path[-1]
-            if event == "end" or not seg.is_type(*types):
-                continue
-
-            if seg is segment:
-                # If the starting segment itself matches the list of types we're
-                # searching for, recursive_crawl() will return it. Skip that.
-                continue
-
+            # Crawl efficiently, don't recurse here. We do that later.
+            # What do we have?
+            # 1. If it's a table reference, work out whether it's to a CTE
+            #    or to an external table.
             if seg.is_type("table_reference"):
-                if not seg.is_qualified() and lookup_cte:
-                    cte = self.lookup_cte(seg.raw, pop=pop)
+                _seg = cast(ObjectReferenceSegment, seg)
```

Comment:
why is this cast necessary?

Reply:
It's so that mypy doesn't complain about the `.is_qualified()` call, which is only available on an object reference.